src/Sylius/Component/Payment/Model/GatewayConfig.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Model;
+
+class GatewayConfig implements GatewayConfigInterface
+{
+    protected mixed $id;
+
+    protected ?string $factoryName = null;
+
+    protected ?string $gatewayName = null;
+
+    /** @var array<string, mixed> $config */
+    protected array $config;
+
+    public function __construct()
+    {
+        $this->config = [];
+    }
+
+    public function getId(): mixed
+    {
+        return $this->id;
+    }
+
+    public function getGatewayName(): ?string
+    {
+        return $this->gatewayName;
+    }
+
+    public function setGatewayName($gatewayName): void
+    {
+        $this->gatewayName = $gatewayName;
+    }
+
+    public function getFactoryName(): ?string
+    {
+        return $this->factoryName;
+    }
```

Comment:
`getFactoryName` will be deprecated one day on the Payum side (BTW: I disagree about removing this getter in Payum):
https://github.com/Payum/Payum/blob/master/src/Payum/Core/Model/GatewayConfigInterface.php#L18
@GSadee WDYT about adding a way like this to always get the factory name evn if one day it will be only available into the `config` array.

```suggestion
    public function getFactoryName(): ?string
    {
        return $this->factoryName ?? $this->config['factory'] ?? null;
    }
```